neuralop/losses/data_losses.py

```diff
@@ -286,6 +286,77 @@ def __call__(self, y_pred, y, h=None, **kwargs):
         return self.rel(y_pred, y, h=h)
 
 
+class DissipativeLoss(object):
+    """Dissipative loss is used to avoid model blow-up and to encourage the model
+    to learn dissipative dynamics (particularly in chaotic systems). The dissipative
+    loss samples points uniformly and randomly in a spherical shell around the
+    attractor of a chaotic system (at user-specified radii) and encourages
+    that a learned model's predictions on this shell scale inwards by some
+    user-specified dissipative rule.
+    
+    Introduced in "Learning Dissipative Dynamics in Chaotic Systems," NeurIPS 2022. 
+    Paper: https://arxiv.org/abs/2106.06898
+
+    Parameters
+    ----------
+    data_loss : function of inputs and outputs
+        Loss function to use for data (e.g., L2 loss)
+    diss_y_rule : function
+        Maps input x on a spherical shell to a desired dissipative y values
+        * E.g., linear scaling: lambda x : b * x for some b
+    loss_weight : float
+        Weighting between the dissipative loss and data loss
+    diss_radii : float tuple
+        Inner and outer radii of shell to sample from
+    out_dim : int
+        Dimension of input/output space (i.e., channels)
+    domain_shape : int tuple
+        Shape of PDE domain (resolution)
+    """
+    def __init__(self, data_loss, diss_y_rule, loss_weight: float, diss_radii: tuple, out_dim: int, domain_shape: tuple):
+        self.data_loss = data_loss
+        self.reduce_dims = data_loss.reduce_dims
+        self.reductions = data_loss.reductions
+
+        self.y_rule = diss_y_rule # callable, maps x on spherical shell to desired dissipative y values
+        self.loss_weight = loss_weight
+        self.radii = diss_radii
+        self.out_dim = out_dim
+        self.dissloss = LpLoss(d=out_dim, reduce_dims=self.reduce_dims, reductions=self.reductions)
+
+        if domain_shape is not None:
+            self.domain_shape = torch.tensor(domain_shape) # for PDE problems, this is a tuple of the domain size (dim1, ..., dimk)
+            self.domain_ndims = torch.prod(self.domain_shape)
+        else:
+            self.domain_shape = None
+    
+    def sample_uniform_spherical_shell(self, shape: tuple): # shape is (batch_size, dim1, ..., dimk, out_dim)
+        npoints = shape[0]
+        shape = shape[1:]
```

Comment:
This is confusing, you are changing the meaning of shape half way through, best to avoid this. Why are the channels at the end? 
You could just do 
batch_size, *spatial_shape, n_channels = shape